nibabel/cifti/cifti.py

```diff
@@ -0,0 +1,868 @@
+
+        This is a class method
+
+        Parameters
+        ----------
+        img : ``spatialimage`` instance
+           In fact, an object with the API of ``spatialimage`` -
+           specifically ``get_data``, ``get_affine``, ``get_header`` and
+           ``extra``.
+        filename : str
+           Filename, implying name to which to save image.
+        '''
+        img = klass.from_image(img)
+        img.to_filename(filename)
+
+    @classmethod
+    def from_image(klass, img):
+        ''' Class method to create new instance of own class from `img`
+
+        Parameters
+        ----------
+        img : ``spatialimage`` instance
+           In fact, an object with the API of ``spatialimage`` -
+           specifically ``get_data``, ``get_affine``, ``get_header`` and
+           ``extra``.
+
+        Returns
+        -------
+        cimg : ``spatialimage`` instance
+           Image, of our own class
+        '''
+        return klass(img._dataobj,
+                     klass.header_class.from_header(img.header),
+                     extra=img.extra.copy())
+
+    def to_filename(self, filename):
+        if not filename.endswith('nii'):
+            ValueError('CIFTI files have to be stored as uncompressed NIFTI2')
+        from ..nifti2 import Nifti2Image
+        from ..nifti1 import Nifti1Extension
+        data = np.reshape(self.data, [1, 1, 1, 1] + list(self.data.shape))
+        header = self.extra
+        extension = Nifti1Extension(32, self.header.to_xml())
+        header.extensions.append(extension)
+        img = Nifti2Image(data, None, header)
+        img.to_filename(filename)
+
+
+class CiftiDenseDataSeries(CiftiImage):
+    """Class to handle Dense Data Series
+    Dense Data Series
+    -----------------
+
+    Intent_code: 3002, NIFTI_INTENT_CONNECTIVITY_DENSE_SERIES
+    Intent_name: ConnDenseSeries
+    File extension: .dtseries.nii
+    AppliesToMatrixDimension 0: series
+    AppliesToMatrixDimension 1: brain models
+
+    This file type represents data points in a series for every vertex and voxel
+    in the mapping.  A row is a complete data series,for a single vertex or
+    voxel in the mapping that applies along the second dimension. A data series
+    is often a timeseries, but it can also represent other data types such as a
+    series of sampling depths along the surface normal from the white to pial
+    surface.  It retains the 't' in dtseries from CIFTI-1 naming conventions.
+    """
+    suffix = '.dtseries.nii'
+
+
+def load(filename):
+    """ Load cifti2 from `filename`
+
+    Parameters
+    ----------
+    filename : str
+        filename of image to be loaded
+
+    Returns
+    -------
+    img : CiftiImage
+        cifti image instance
+
+    Raises
+    ------
+    ImageFileError: if `filename` doesn't look like cifti
+    IOError : if `filename` does not exist
+    """
+    from ..nifti2 import load as Nifti2load
+    return Nifti2load(filename).as_cifti()
```

Comment:
I'd rather see a `Ciftiload` function that calls `Nifti2load`, then adds extra CIFTI-specific functionality currently in `as_cifti`.